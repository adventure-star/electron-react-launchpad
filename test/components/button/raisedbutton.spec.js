import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import RaisedButton from '../../../src/renderer/components/button/raisedbutton'

function testFunction() {
    // DO NOTHING

    return "";
}

function setupStandard() {
    const renderer = ReactTestRenderer.create(<RaisedButton onClick={testFunction} className='testClass'>Test button text</RaisedButton>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Button - RaisedButton component', () => {
    it('should be of type button', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('button');
    });

    it('should have only the classes "btn" and "testClass"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('btn testClass');
    });

    it('should have an onClick function', () => {
        const { component } = setupStandard();
        expect(component.props.onClick).toBe(testFunction);
    });

    it('children should be the text "Test content"', () => {
        const { component } = setupStandard();
        expect(component.children[0]).toBe('Test button text');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
