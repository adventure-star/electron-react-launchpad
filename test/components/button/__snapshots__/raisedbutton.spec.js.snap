// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - RaisedButton component Matches/Renders a snapshot 1`] = `
<button
  className="btn testClass"
  onClick={[Function]}
>
  Test button text
</button>
`;
